kroxylicious-filters/kroxylicious-authorization/src/test/java/io/kroxylicious/filter/authorization/AuthorizationFilterTest.java

```diff
@@ -96,13 +99,24 @@ void authorization(ScenarioDefinition definition) {
         if (!mockUpstream.isFinished()) {
             throw new IllegalStateException("test has finished, but mock responses are still queued");
         }
+
+        assertOnlyDenialsLogged();
```

Comment:
I wonder if we need a unit test where we know explicitly which resources will be denied and which will be allowed. That would allow us to write a more specific assertion that all the denied things were logged as denied. 

I also think we should log the `allowed` decisions, but at a lower log level since they will usually be very noisy. Doing that would enable asserting that a decision was logged exactly once for each Action.

Reply:
> I wonder if we need a unit test where we know explicitly which resources will be denied and which will be allowed. That would allow us to write a more specific assertion that all the denied things were logged as denied.

I think thats getting much more interesting, but I'm not sure I want to pull it into the scope of this fix.